omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_9/LocationAttributeSubSearchHandler1_9.java

```diff
@@ -0,0 +1,60 @@
+package org.openmrs.module.webservices.rest.web.v1_0.search.openmrs1_9;
+
+import org.openmrs.Location;
+import org.openmrs.LocationAttribute;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.api.SearchConfig;
+import org.openmrs.module.webservices.rest.web.resource.api.SearchQuery;
+import org.openmrs.module.webservices.rest.web.resource.api.SubSearchHandler;
+import org.openmrs.module.webservices.rest.web.resource.impl.EmptySearchResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Component
+public class LocationAttributeSubSearchHandler1_9 implements SubSearchHandler {
+
+    private final static SearchConfig SEARCH_CONFIG = new SearchConfig(
+            "default",
+            RestConstants.VERSION_1 + "/location/attribute",
+            Arrays.asList("1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*"),
+            new SearchQuery.Builder("Allows you to find attributes by attribute type")
+                    .withRequiredParameters("attributeType")
+                    .build()
+    );
+
+    @Override
+    public SearchConfig getSearchConfig() {
+        return SEARCH_CONFIG;
+    }
+
+    @Override
+    public PageableResult search(RequestContext context) throws ResponseException {
+        throw new UnsupportedOperationException("Cannot search for location attributes without parent location");
+    }
+
+    @Override
+    public PageableResult search(String parentUuid, RequestContext context) throws ResponseException {
+        String attributeType = context.getParameter("attributeType");
+        Location parentLocation = Context.getLocationService().getLocationByUuid(parentUuid);
+
+        if(attributeType == null || parentUuid == null){
+            return new EmptySearchResult();
+        }
+
+        List<LocationAttribute> results = new ArrayList<LocationAttribute>();
+        for(LocationAttribute activeAttribute : parentLocation.getActiveAttributes()){
+            if(activeAttribute.getAttributeType().getUuid().equals(attributeType)){
+                results.add(activeAttribute);
+            }
+        }
+        return new NeedsPaging<LocationAttribute>(results, context);
```

Comment:
This should be an instance of AlreadyPaged since there are no more results to be fetched